### PR TITLE
Add Container task

### PR DESCRIPTION
**Pull request by evdokim**
Adds support for ContainerTask: https://docs.flyte.org/projects/flytekit/en/latest/generated/flytekit.ContainerTask.html

Container tasks will allow to execute arbitrary container as Flyte task.

# TL;DR
Adds `ContainerTask` allowing to execute arbitrary container as a Flyte task. Such functionality already exists in Python SDK: https://docs.flyte.org/projects/flytekit/en/latest/generated/flytekit.ContainerTask.html 

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
[ContainerTask](https://docs.flyte.org/projects/flytekit/en/latest/generated/flytekit.ContainerTask.html) functionality has been ported with minimal changes to the API. Container tasks will have its own registar therefore minimum changes were made to existing code for runnable tasks.

An example of a container task added here: https://github.com/evdokim/flytekit-java/blob/container-task/flytekit-examples/src/main/java/org/flyte/examples/HelloContainerTask.java

## Tracking Issue
No issues has been created

## Follow-up issue
No follow up issues

### CODE REVIEW

File: flytekit-java/src/main/java/org/flyte/flytekit/SdkRunnableTaskRegistrar.java [hunk 1]
```diff
@@ -86,43 +86,7 @@ public String getName() {
 
     @Override
     public Resources getResources() {
-      SdkResources sdkResources = sdkTask.getResources();
```
Comment (evdokim) [inline flytekit-java/src/main/java/org/flyte/flytekit/SdkRunnableTaskRegistrar.java:89]: moved to `SdkResources.java`

**Review by narape — CHANGES_REQUESTED** [review]
@evdokim first, thanks for the contribution. Here is one bug and some minor changes

File: flytekit-api/src/main/java/org/flyte/api/v1/ContainerTask.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  List<String> getArgs();
+
+  /** Specifies container command. Example: {@code List.of("java", "-Xmx1G")}. */
+  List<String> getCommand();
+
+  /** Specifies container environment variables. */
+  List<KeyValuePair> getEnv();
+
+  default String getType() {
+    return "raw-container".toLowerCase(Locale.ROOT);
```
[the remark this answers is not in the document]
Comment (evdokim) [reply flytekit-api/src/main/java/org/flyte/api/v1/ContainerTask.java:43]: Good catch, I don't even remember why I added this conversion

File: flytekit-java/src/main/java/org/flyte/flytekit/SdkResources.java [hunk 3]
```diff
@@ -55,6 +75,27 @@ public static SdkResources empty() {
     return EMPTY;
   }
 
+  public Resources toResources() {
```
[the remark this answers is not in the document]
Comment (evdokim) [reply flytekit-java/src/main/java/org/flyte/flytekit/SdkResources.java:78]: Agree, `toIdl()` looks more cleaner

**Comment by evdokim** [comment]
@narape thank you for the review, I addressed your comments, could you take a look again?

**Review by narape — APPROVED** [review]

**Comment by narape** [comment]
Snyk problem was fixed in https://github.com/flyteorg/flytekit-java/pull/113

**Pull request merged**